tycho-core/src/main/java/org/eclipse/tycho/core/maven/TychoWorkspaceReader.java

```diff
@@ -68,110 +58,149 @@ public WorkspaceRepository getRepository() {
+        //  added by tycho itself.
+        return findMavenArtifact(artifact);
     }
 
     @Override
-    public List<String> findVersions(Artifact artifact) {
+    public List<String> findVersions(final Artifact artifact) {
         return Collections.emptyList();
     }
 
     @Override
-    public Model findModel(Artifact artifact) {
+    public Model findModel(final Artifact artifact) {
         if (artifact.getGroupId().startsWith(TychoConstants.P2_GROUPID_PREFIX)) {
-            logger.debug("Find the model for: " + artifact);
-            //TODO due to a bug in maven we can not use this here... see  Tycho issue #1388
-            //see https://issues.apache.org/jira/browse/MNG-7544
-//            return getP2Model(artifact);
+            logger.debug("Find the model for " + artifact);
+            // TODO: due to a bug in Maven we can not use this here.
+            //  See Tycho issue #1388
+            //  See https://issues.apache.org/jira/browse/MNG-7544
+            // return getP2Model(artifact);
+        }
+
+        return null;
+    }
+
+    private File findPomArtifact(final Artifact artifact) {
+        logger.debug("Find the POM file for " + artifact);
+        final File pomFile = getFileForArtifact(artifact);
+
+        if (pomFile.isFile()) {
+            return pomFile;
+        }
+
+        pomFile.getParentFile().mkdirs();
+
+        try {
+            modelWriter.write(pomFile, new HashMap<>(), getP2Model(artifact));
+            return pomFile;
+        } catch (final IOException e) {
+            logger.debug("Cannot write the POM model", e);
         }
+
+        return null;
+    }
+
+    private File findP2Artifact(final Artifact artifact) {
+        // Attempt a fast lookup first
+        final File cachedFile = getFileForArtifact(artifact);
+
+        if (cachedFile.isFile()) {
+            return cachedFile;
+        }
+
+        final MavenSession session = legacySupport.getSession();
+
+        if (session != null) {
+            final MavenProject currentProject = session.getCurrentProject();
+            final ReactorProject reactorProject = DefaultReactorProject.adapt(currentProject);
+            final Optional<DependencyArtifacts> dependencyMetadata =
+                    TychoProjectUtils.getOptionalDependencyArtifacts(reactorProject);
+
+            if (dependencyMetadata.isPresent()) {
+                logger.debug("Attempting to resolve " + artifact + " for project " + currentProject + "...");
+
+                for (final ArtifactDescriptor descriptor : dependencyMetadata.get().getArtifacts()) {
+                    final ArtifactKey artifactKey = descriptor.getKey();
+                    final String artifactId = artifactKey.getId();
+                    final String version = artifactKey.getVersion();
+
+                    if (artifactId.equals(artifact.getArtifactId()) && version.equals(artifact.getVersion())) {
```

Comment:
Should we check if the artifact type matches?  
Examples:
```
p2.eclipse.plugin  = eclipse-plugin
p2.eclipse.feature = eclipse-feature
```

Reply:
Features are usually nothing one can "resolve" in maven (at least at the moment), but technically it might be good to check the artifact type.

Reply:
@laeubi 

> are usually nothing one can "resolve" in maven

Guess I'm the first one resolving (I need the JAR to be on disk) feature bundles then 😄   
Ok, I'll add the check.

Reply:
Done.